framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -385,6 +487,24 @@ public void insertValue(JavaExpression expr, @Nullable V value) {
             // Expressions containing unknown expressions are not stored.
             return;
         }
+        if (!(permitNondeterministic || expr.isDeterministic(analysis.getTypeFactory()))) {
+            // Nondeterministic expressions may not be stored.
+            // (They are likely to be quickly evicted, as soon as a side-effecting method is
+            // called.)
+            return;
+        }
+        insertValueImpl(expr, value);
+    }
+
+    /**
+     * Helper method for {@link #insertValue(JavaExpression, CFAbstractValue)} and {@link
+     * insertValuePermitNondeterministic}. This implementation does the real work. The expression
+     * will definitely be inserted.
```

Comment:
`The expression will definitely be inserted.`

This isn't true. Only some classes of JavaExpressions are inserted and even then only if `sequentialSemantics` is true.